demos-europe/demosplan-core
# feat (refs DPLAN-603, DPLAN-1901): Introduce rate limits against brute force

**Ticket:** https://demoseurope.youtrack.cloud/issue/DPLAN-1901/pentest-Bruteforce-Login-3.0-Identity-Management-Testing-3.4-Testing-for-Account-Enumeration-and-Guessable-User-Account-PS-TW

**Ticket:** https://demoseurope.youtrack.cloud/issue/DPLAN-1900/10-Business-Logic-Testing-10.5-Test-Number-of-Times-a-Function-Can-be-Used-Limits-PS-TW-2021-0105

Symfony rate limiter is used as counter measure against brute force attacks. It is applied to login, register as organisation or user and anonymous statement submit.


### How to review/test
There are 4 parts:

1. Try to Login as the same user with a wrong password more than 5 times a minute. A distinct error should appear. When you then try to login within the minute with correct credentials you should not be able to log in.
2. Try to register as a citizen. You should only be able to send 2 requests per hour.
3. Try to register as an organisation. You should only be able to send 2 requests per hour.
4. Try to submit an statement via public detail page as a guest. You should only be able to send 4 Statements per 15 Minutes

### Linked PRs (optional)


### Tasks (optional)


### PR Checklist


Delete the checkbox if it doesn't apply/isn't necessary.
